rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -1016,21 +1015,173 @@ void removeSignaturesFromTransactionWithP2shMultiSigInputs_whenNotAllInputsHaveP
+        Federation federation = P2shP2wshErpFederationBuilder.builder()
+            .withMembersBtcPublicKeys(federationBtcKeys)
+            .build();
+
+        BtcTransaction transaction = PegoutTransactionBuilder.builder()
+            .withActiveFederation(federation)
+            .build();
+        BtcTransaction transactionBeforeSigning = new BtcTransaction(btcMainnetParams, transaction.bitcoinSerialize());
+
+        List<TransactionInput> inputs = transaction.getInputs();
+        for (TransactionInput input : inputs) {
+            Coin inputValue = input.getValue();
+            BitcoinTestUtils.signWitnessTransactionInputFromP2shMultiSig(
+                transaction,
+                inputs.indexOf(input),
+                inputValue,
+                federationBtcKeys
+            );
+        }
+
+        // Make sure the transaction has signatures
+        assertNotEquals(transactionBeforeSigning, transaction);
+
+        // act
+        BitcoinUtils.removeSignaturesFromMultiSigTransaction(transaction);
+
+        // assert
+        assertEquals(transactionBeforeSigning, transaction);
+
+        Script federationRedeemScript = federation.getRedeemScript();
+        List<TransactionInput> transactionInputs = transaction.getInputs();
+        for (TransactionInput input : transactionInputs) {
+            int inputIndex = transactionInputs.indexOf(input);
+            TransactionWitness witness = transaction.getWitness(inputIndex);
+
+            assertWitnessScriptWithoutSignaturesHasProperFormat(witness, federationRedeemScript);
+            assertSegwitScriptSigContainsHashedRedeemScript(input.getScriptSig(), federationRedeemScript);
+        }
+    }
+
+    @Test
+    void removeSignaturesFromMultiSigTransaction_withLegacyAndSegwitInputs_shouldReturnExpectedTx() {
+        // arrange
+        List<BtcECKey> p2shFederationBtcKeys = BitcoinTestUtils.getBtcEcKeys(9);
+        Federation p2shFederation = P2shErpFederationBuilder.builder()
+            .withMembersBtcPublicKeys(p2shFederationBtcKeys)
+            .build();
+
+        List<BtcECKey> p2wshFederationBtcKeys = BitcoinTestUtils.getBtcEcKeys(20);
+        Federation p2wshFederation = P2shP2wshErpFederationBuilder.builder()
+            .withMembersBtcPublicKeys(p2wshFederationBtcKeys)
+            .build();
+
+        BtcTransaction transaction = new BtcTransaction(btcMainnetParams);
+        TransactionInput legacyInput = new TransactionInput(
+            btcMainnetParams,
+            null,
+            new byte[]{},
+            new TransactionOutPoint(btcMainnetParams, 0, BitcoinTestUtils.createHash(3)),
+            Coin.COIN
+        );
+        TransactionInput segwitInput = new TransactionInput(
+            btcMainnetParams,
+            null,
+            new byte[]{},
+            new TransactionOutPoint(btcMainnetParams, 1, BitcoinTestUtils.createHash(6)),
+            Coin.COIN.multiply(2)
+        );
```

Comment:
```suggestion
        int legacyInputIndex = 0;
        int segwitInputIndex = 1;
        
        TransactionInput legacyInput = new TransactionInput(
            btcMainnetParams,
            null,
            new byte[]{},
            new TransactionOutPoint(btcMainnetParams, legacyInputIndex, BitcoinTestUtils.createHash(3)),
            Coin.COIN
        );
        TransactionInput segwitInput = new TransactionInput(
            btcMainnetParams,
            null,
            new byte[]{},
            new TransactionOutPoint(btcMainnetParams, segwitInputIndex, BitcoinTestUtils.createHash(6)),
            Coin.COIN.multiply(2)
        );
```

wdyt? especially to reuse it below